community/codegen/src/main/java/org/neo4j/codegen/CodeGenerationStrategy.java

```diff
@@ -76,8 +87,10 @@ public void addByteCodeVisitor( ByteCodeVisitor visitor )
         }
 
         CodeGenerator generateCode( ClassLoader loader, CodeGeneratorOption[] options )
+                throws CodeGenerationNotSupportedException
         {
-            CodeGenerator generator = strategy.generateCode( loader, options );
+            CodeGenerator generator;
+            generator = strategy.generateCode( loader, options );
```

Comment:
So do I... How did that come to look this way?

Right... I tried handling the exception here, by wrapping and re-throwing, but then I just changed the exception hierarchy instead...